e2e_tests/tests/test_system.py

```diff
@@ -595,6 +595,64 @@ def test_disable_and_enable_slots() -> None:
     subprocess.check_call(command)
 
 
+@pytest.mark.e2e_cpu  # type: ignore
+def test_drain_agent() -> None:
+    """
+    Start an experiment, `disable --drain` the agent once the trial is running,
+    make sure the experiment still finishes, but the new ones won't schedule.
+    """
+
+    def _fetch_slots() -> List[Dict[str, Any]]:
+        command = [
+            "det",
+            "-m",
+            conf.make_master_url(),
+            "slot",
+            "list",
+            "--json",
+        ]
+        output = subprocess.check_output(command).decode()
+        slots = cast(List[Dict[str, str]], json.loads(output))
+        assert len(slots) == 1
+        return slots
+
+    slots = _fetch_slots()
+    agent_id = slots[0]["agent_id"]
+
+    experiment_id = exp.create_experiment(
+        conf.fixtures_path("no_op/single-medium-train-step.yaml"),
+        conf.fixtures_path("no_op"),
+        None,
+    )
+    exp.wait_for_experiment_state(experiment_id, "ACTIVE")
+    exp.wait_for_experiment_active_workload(experiment_id)
+    exp.wait_for_experiment_workload_progress(experiment_id)
+
+    command = ["det", "-m", conf.make_master_url(), "agent", "disable", "--drain", agent_id]
+    subprocess.check_call(command)
+
+    exp.wait_for_experiment_state(experiment_id, "COMPLETED")
```

Comment:
I think we should also check the experiment is @ the last step we expected it to complete.